From 69078f41b140efeb594b1ef6adccb672cd53a3dd Mon Sep 17 00:00:00 2001
From: Fred Fan <r01011@freescale.com>
Date: Tue, 20 May 2008 11:33:10 +0800
Subject: [PATCH] ENGR00076754 Fix mxc_rtc can not connect with PMIC RTC on MX35 3stack

Move mxc_rtc to device_initcall_sync to make mxc_rtc initialled after pmic_rtc.

Signed-off-by: Fred Fan <r01011@freescale.com>
---
 drivers/rtc/rtc-mxc.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/rtc/rtc-mxc.c b/drivers/rtc/rtc-mxc.c
index c3997d7..a0735ed 100644
--- a/drivers/rtc/rtc-mxc.c
+++ b/drivers/rtc/rtc-mxc.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2004-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -120,7 +120,8 @@ struct rtc_plat_data {
  * @ingroup RTC
  */
 
-#if defined (CONFIG_MXC_PMIC_SC55112_RTC) || defined (CONFIG_MXC_MC13783_RTC)
+#if defined(CONFIG_MXC_PMIC_SC55112_RTC) || defined(CONFIG_MXC_MC13783_RTC) ||
+defined(CONFIG_MXC_MC9SDZ60_RTC)
 #include <asm/arch/pmic_rtc.h>
 #else
 #define pmic_rtc_get_time(args)	MXC_EXTERNAL_RTC_NONE
@@ -807,7 +808,7 @@ static void __exit mxc_rtc_exit(void)
 
 }
 
-module_init(mxc_rtc_init);
+device_initcall_sync(mxc_rtc_init);
 module_exit(mxc_rtc_exit);
 
 MODULE_AUTHOR("Freescale Semiconductor, Inc.");
-- 
1.5.4.4

